### DISCUSSION TITLE
Is it possible to replace list values when `extend`-ing another toml file

### DISCUSSION THREAD

**Discussion by brki** [astral-sh/ruff#13393] [Q&A]
For example, with `parent-ruff.toml`:

```
line-length = 120

[lint]
select = ["E", "W","F"]
ignore = ["E402"]
```

And `project-ruff.toml`:
```
extend = "./parent-ruff.toml"

line-length = 80

[lint]
ignore = ["E501"]
```

I would like it if the final `lint.ignore` value is `["E501"]`.
The actual value seems to be `["E402", "E501"]`.

The scalar `line-length` value takes the value from `project-ruff.toml`, but the list value seems to have the values combined, like a set union.

**Comment by MichaReiser** [comment]
The semantics on how values are combined depends on the setting. I don't think there's a way to unset values in the `ignore` list. But you can re-select the rule using `extend-select = ["E402"]`

**Reply by Avasam** [reply]
"unignoring" is basically the same as "reselecting" in most cases. (The difference is there's no "go back to default", but Ruff doesn't have many default rules anyway).

This should work fine with eventual shared configs, which would"ve been my concern.
